aws/session/internal/ini/number_helper.go

```diff
@@ -0,0 +1,135 @@
+package ini
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+// numberHelper is used to dictate what format a number is in
+// and what to do for negative values. Since -1e-4 is a valid
+// number, we cannot just simply check for duplicate negatives.
+type numberHelper struct {
+	binary  bool
```

Comment:
Suggest using a `type` of number to switch between later instead of multiple bools